tests/integration/test_duckdb_tpch.py

```diff
@@ -0,0 +1,108 @@
+class TestDuckDBConsumer:
+    """
+    Test Class for testing Substrait using Acero as a consumer.
+    """
+
+    @staticmethod
+    @pytest.fixture(scope="class", autouse=True)
+    def setup_teardown_class(request):
+        cls = request.cls
+
+        cls.db_connection = duckdb.connect()
+        cls.db_connection.execute("INSTALL substrait")
+        cls.db_connection.execute("LOAD substrait")
+        cls.consumer = DuckDBConsumer()
+        cls.utils = SubstraitUtils()
+        cls.created_tables = set()
+
+        yield
+
+        cls.db_connection.close()
+
+    @custom_parametrization(TPCH_QUERY_TESTS)
+    def test_substrait_query(
+        self,
+        test_name: str,
+        file_names: list,
+        sql_query: str,
+        substrait_query: str,
+        sort_results: bool = False,
+    ) -> None:
+        """
+        1.  Load all the parquet files into DuckDB as separate tables.
+        2.  Format the SQL query to work with DuckDB by inserting all the table names.
+        3.  Execute the SQL on DuckDB.
+        4.  Run the substrait query plan.
+        5.  Compare substrait query plan results against the results of
+            running the SQL on DuckDB.
+
+        Parameters:
+            test_name:
+                Name of test.
+            file_names:
+                List of parquet files.
+            sql_query:
+                SQL query.
+            sort_results:
+                Whether to sort the results before comparison.
+        """
+
+        parquet_file_paths = self.utils.get_full_path(file_names)
+
+        # Load the parquet files into DuckDB and return all the table names as a list
+        table_names = self.consumer.load_tables_from_parquet(
+            self.db_connection, self.created_tables, file_names, parquet_file_paths
+        )
+
+        # Format the sql query by inserting all the table names
+        sql_query = sql_query.format(*table_names)
+
+        # Convert the SQL into a substrait query plan and run the plan.
+        subtrait_query_result_tb = self.consumer.run_substrait_query(
+            self.db_connection, sql_query
+        )
+
+        # Calculate results to verify against by running the SQL query on DuckDB
+        duckdb_sql_result_tb = self.db_connection.query(f"{sql_query}").arrow()
+
+        col_names = [x.lower() for x in subtrait_query_result_tb.column_names]
+        exp_col_names = [x.lower() for x in duckdb_sql_result_tb.column_names]
+
+        # Sort results by specified column names
+        if sort_results:
+            subtrait_sort_col = subtrait_query_result_tb.column_names[0]
+            subtrait_query_result_tb = self.utils.arrow_sort_tb_values(
+                subtrait_query_result_tb, sortby=[subtrait_sort_col]
+            )
+            duckdb_sort_col = duckdb_sql_result_tb.column_names[0]
+            duckdb_sql_result_tb = self.utils.arrow_sort_tb_values(
+                duckdb_sql_result_tb, sortby=[duckdb_sort_col]
+            )
```

Comment:
Just to clarify, I do this sorting on column names, because internally we do test the consumption of ibis substrait plans vs ibis query results, and ibis does not output the results in the order set by the SQL query. Hence, this does not seem to be necessary here.